docs/book/src/tasks/automated-machine-management/machine_deletions.md

```diff
@@ -0,0 +1,127 @@
+
+Note: There are cases where Node drain, wait for volume detach and Node deletion is skipped. For these please take a look at the 
+implementation of the [`isDeleteNodeAllowed` function](https://github.com/kubernetes-sigs/cluster-api/blob/v1.8.0/internal/controllers/machine/machine_controller.go#L346).
+
+## Node drain
+
+This section describes details of the Node drain process.
+
+Node drain can be broken down into the following phases:
+* Node is cordoned (i.e. the `Node.spec.unschedulable` field is set, which leads to the `node.kubernetes.io/unschedulable:NoSchedule` taint being added to the Node)
+  * This prevents that Pods that already have been evicted are rescheduled to the same Node. Please only tolerate this taint 
+    if you know what you are doing! Otherwise it can happen that the Machine controller is stuck continuously evicting the same Pods. 
+* Machine controller calculates the list of Pods that should be evicted. These are all Pods on the Node, except:
+  * Pods belonging to an existing DaemonSet (orphaned DaemonSet Pods have to be evicted as well)
+  * Mirror Pods, i.e. Pods with the `kubernetes.io/config.mirror` annotation (usually static Pods managed by kubelet, like `kube-apiserver`)
+* If there are no (more) Pods that have to be evicted and all Pods that have been evicted are gone, Node drain is completed
+* Otherwise an eviction will be triggered for all Pods that have to be evicted. There are various reasons why an eviction call could fail:
+  * The eviction would violate a PodDisruptionBudget, i.e. not enough Pod replicas would be available if the Pod would be evicted
+  * The namespace is in terminating, in this case the `kube-controller-manager` is responsible for setting the `.metadata.deletionTimestamp` on the Pod
+  * Other errors, e.g. a connection issue when calling the eviction API at the workload cluster
+* Please note that when an eviction goes through, this only means that the `.metadata.deletionTimestamp` is set on the Pod, but the 
+  Pod also has to be terminated and the Pod object has to go away for the drain to complete.
+* These steps are repeated every 20s until all relevant Pods have been drained from the Node
+
+Special cases:
+* If the Node doesn't exist anymore, Node drain is entirely skipped
+* If the Node is `unreachable` (i.e. the Node `Ready` condition is in status `Unknown`):
+  * Pods with `.metadata.deletionTimestamp` more than 1s in the past are ignored
+  * Pod evictions will use 1s `GracePeriodSeconds`, i.e. the `terminationGracePeriodSeconds` field from the Pod spec will be ignored.
+  * Note: PodDisruptionBudgets are still respected, because both of these changes are only relevant if the call to trigger the Pod eviction goes through.
+    But Pod eviction calls are rejected when PodDisruptionBudgets would be violated by the eviction.
+
+### Observability
```

Comment:
Since this doc are user oriented, I will first show the example condition, and only after the example logs (logs are for SRE only)